plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java

```diff
@@ -86,7 +88,9 @@ public void createView(ConnectorSession session, SchemaTableName schemaViewName,
 
         Optional<io.trino.plugin.hive.metastore.Table> existing = metastore.getTable(schemaViewName.getSchemaName(), schemaViewName.getTableName());
         if (existing.isPresent()) {
-            if (!replace || !isPrestoView(existing.get())) {
+            Table existingTable = existing.get();
+            boolean shouldReplace = isPrestoView(existingTable) && !isTrinoMaterializedView(existingTable.getTableType(), existingTable.getParameters());
```

Comment:
> `isPrestoView` returns true for both regular views and Materialized
Views, so it could have been possible for a view to replace a MV.

@raunaqmorarka why is that?

doesn't seem practical
on the code level, the MVs are handled pretty separately, so why would we want to conflate them?

Reply:
I'm not sure if that was on purpose or an accident, but we've been setting the "presto view" flag for MVs since the initial implementation https://github.com/trinodb/trino/pull/4832/files#diff-0a3bb720afd0370525ff486d8d55d6140e766028d20c95c37e8d60f801790881R765
Maybe it was a shortcut to ensure that MVs get listed in the metadata listing for views.
I agree that a clear separation would be better going forward.

Reply:
@alexjo2144 would it be possible to fix isPrestoView so that it doesn't return true for MVs?

Can you run this as a separate PR, potentially obsoleting this one?

Reply:
i am following-up on this in https://github.com/trinodb/trino/pull/18570